lib/front_desk.rb

```diff
@@ -0,0 +1,187 @@
+    attr_accessor :all_rooms, :reservations_record, :block_reservations_only, :reserved_rooms_in_blocks
+
+    def initialize
+      @all_rooms = [*(1..20)] # => makes array with elements from 1 to 20
+
+      @reservations_record = Array.new # => stores all reservation objects
+
+      @block_reservations_only = Array.new # Stores all block reservations
+
+      @reserved_rooms_in_blocks = Array.new # stores the reservations of rooms reserved in blocks
+    end
+
+    def all_rooms_array
+      return @all_rooms
+    end
+
+    def reservations_record_array
+      return @reservations_record
+    end #
+
+    def block_reservations_array
+      return @block_reservations_only
+    end
+
+    def reserved_rooms_in_blocks_array
+      return @reserved_rooms_in_blocks
+    end
+
+    def reservation_template(new_booking_ref, new_number_of_rooms, new_first_room_number, new_check_in, new_check_out)
+      booking = Hotel::Reservation.new(
+        booking_ref: new_booking_ref,
+        number_of_rooms: new_number_of_rooms,
+        first_room_number: new_first_room_number,
+        check_in: new_check_in,
+        check_out: new_check_out,
+        total_cost: @total_cost,
+        room_blocks: @room_blocks,
+      )
+      return booking
+    end
+
+    def make_reservation(new_booking_ref, new_number_of_rooms, new_first_room_number, new_check_in, new_check_out)
+      if (@reservations_record.length == 0)
+        booking = reservation_template(new_booking_ref, new_number_of_rooms, new_first_room_number, new_check_in, new_check_out)
+      else
+        for i in (0...@reservations_record.length)
+          old_first_room_no = @reservations_record[i].first_room_number
+          old_last_room_no = @reservations_record[i].first_room_number + @reservations_record[i].number_of_rooms - 1
+          new_last_room_no = new_first_room_number + new_number_of_rooms - 1
+
+          if ((old_first_room_no >= new_first_room_number && old_first_room_no <= new_last_room_no) ||
+              (old_last_room_no >= new_first_room_number && old_last_room_no <= new_last_room_no) ||
+              ((old_first_room_no <= new_first_room_number) && (old_first_room_no <= new_last_room_no) && (old_last_room_no >= new_first_room_number) && (old_last_room_no >= new_last_room_no)))
+            if ((new_check_in >= @reservations_record[i].check_in) && (new_check_in < @reservations_record[i].check_out)) ||
+               ((new_check_out >= @reservations_record[i].check_in) && (new_check_out < @reservations_record[i].check_out))
+              raise ArgumentError, "No BLOCK 4 U. This room is unavailable."
+            else
+              booking = reservation_template(new_booking_ref, new_number_of_rooms, new_first_room_number, new_check_in, new_check_out)
+            end
+          else
+            booking = reservation_template(new_booking_ref, new_number_of_rooms, new_first_room_number, new_check_in, new_check_out)
+          end
+        end
+      end
```

Comment:
Your variable names and your complex logic here are overwhelming and confusing, and difficult to read. This would be great pulled into helper methods.